ocular/src/chain_info.rs

```diff
@@ -133,39 +131,55 @@ impl ChainInfo {
         if let Some(endpoint) = endpoints.choose(&mut thread_rng()) {
             return Ok(endpoint.to_string());
         } else {
-            Err(ChainInfoError::RpcEndpoint(
-                "no available RPC endpoints".to_string(),
-            ))
+            Err(RpcError::UnhealthyEndpoint("no available RPC endpoints".to_string()).into())
         }
     }
 
     pub async fn get_rpc_endpoints(&self) -> Result<Vec<String>, ChainInfoError> {
         let mut endpoints = self.get_all_rpc_endpoints();
         if endpoints.is_empty() {
-            return Err(ChainInfoError::RpcEndpoint(
+            return Err(RpcError::UnhealthyEndpoint(
                 "no valid endpoint found. endpoints must use http or https.".to_string(),
-            ));
+            )
+            .into());
         }
 
         // this is not very efficient but i was getting annoyed trying to figure
         // out how to do filtering with an async method
         for (i, ep) in endpoints.clone().iter().enumerate() {
-            if !chain_registry::is_healthy_rpc(ep.as_str()).await {
+            if is_healthy_rpc(ep.as_str()).await.is_err() {
                 endpoints.remove(i);
             }
         }
         if endpoints.is_empty() {
-            return Err(ChainInfoError::RpcEndpoint(
+            return Err(RpcError::UnhealthyEndpoint(
                 "no healthy endpoint found (connections could not be established)".to_string(),
-            ));
+            )
+            .into());
         }
 
         Ok(endpoints)
     }
 }
 
 pub async fn get_cosmoshub_info() -> Result<ChainInfo, ChainInfoError> {
-    chain_registry::get_chain("cosmoshub").await.map_err(|r| r.into())
+    chain_registry::get_chain("cosmoshub")
+        .await
+        .map_err(|r| r.into())
+}
+
+pub async fn is_healthy_rpc(endpoint: &str) -> Result<(), ChainInfoError> {
```

Comment:
Maybe we could add a couple lightweight tests around this method, get_cosmoshub_info, and get_rpc_endpoints, & query_latest_height?

Reply:
done.  ended up finding some issues in the core chain registry and config building process so this was good to go ahead and knock out.